### PR TITLE
[RFC] add timestamp to mqtt message

### PR DESCRIPTION
**Pull request by lolorc**
see commit msgs.

graph sensors with update period >= 5s
g1

graph sensors with update period > 3640s
g2

I clearly have issues with some of my devices :-)

### CODE REVIEW

File: lib/controller.js [hunk 1]
```diff
@@ -555,7 +563,15 @@ class Controller {
         if (cacheState) {
             // Add cached state to payload
             if (this.state.exists(deviceID)) {
-                messagePayload = objectAssignDeep.noMutate(this.state.get(deviceID), payload);
+                const state = this.state.get(deviceID);
+                if (mergeCacheState) {
```
Comment (Koenkk) [inline lib/controller.js]: This is already done with the [`cache_state` option](https://github.com/Koenkk/zigbee2mqtt/wiki/Configuration)

File: lib/controller.js [hunk 2]
```diff
@@ -555,7 +563,15 @@ class Controller {
...
+                    messagePayload = objectAssignDeep.noMutate(state, payload);
+                }
+                if (addTimestamp) {
```
Comment (Koenkk) [inline lib/controller.js]: This can be moved to line 593

File: lib/controller.js [hunk 3]
```diff
@@ -308,6 +313,9 @@ class Controller {
                     payload.linkquality = message.linkquality;
                 }
 
+                if (addTimestamp) {
```
Comment (Koenkk) [inline lib/controller.js]: Not needed anymore after https://github.com/Koenkk/zigbee2mqtt/pull/450/files#r223127348

File: lib/controller.js (hunk 1 above)
Comment (lolorc) [reply lib/controller.js]: in my case, I'd like to use the cache to store the previous timestamp, but I don't want other attributes to merged. I'm not changing the default behavior, and that might be an useless option.
up to you :-)

**Comment by lolorc** [comment]
feature merged in dev

**Pull request closed without merging**